### DISCUSSION TITLE
i want to query log and active sessions. any idea ?

### DISCUSSION THREAD

**Discussion by r33285** [vladimirs-git/fortigate-api#13] [Ideas]
I want to query log and active sessions. any idea on when that can be included ?

**Comment by vladimirs-git** [comment]
You can get sessions and logs by SSH or by REST API as in the following example

code :: python
    
    from fortigate_api import FortigateAPI

    fgt = FortigateAPI(host="host", username="username", password="password")


    # Get sessions by SSH
    sessions = fgt.ssh.send_command("get system session list")
    print(sessions)


    # Get sessions by REST API
    response = fgt.rest.get(url="api/v2/monitor/firewall/session?vdom=root&start=0&count=1")
    print(response["details"])


    # Get logs by SSH
    log_settings = [
        "execute log filter reset",
        "execute log filter category traffic",
        "execute log filter device memory",
        "execute log filter view-lines 100",
    ]
    fgt.ssh.send_config_set(log_settings)
    output = fgt.ssh.send_command("execute log display")
    print(output)


    # Get logs by REST API
    data = fgt.rest.get(url="/api/v2/log/memory/traffic/forward")
    print(data)
    

Unfortunately, I have no experience with getting sessions by the REST API, you need to know the parameters to get interested sessions. I just fond some example in other project https://github.com/acidjunk/fortigate-python-wrapper/blob/master/examples/fos_api.py#L271


Many other tutorials you can find in [./examples](https://github.com/vladimirs-git/fortigate-api/tree/main/examples)
Please star this project if you found it useful.
